client/rest-high-level/src/test/java/org/elasticsearch/client/PingAndInfoIT.java

```diff
@@ -19,13 +19,33 @@
 
 package org.elasticsearch.client;
 
+import org.elasticsearch.action.main.MainResponse;
+
 import java.io.IOException;
+import java.util.Map;
 
 public class PingAndInfoIT extends ESRestHighLevelClientTestCase {
 
     public void testPing() throws IOException {
         assertTrue(highLevelClient().ping());
     }
 
-    //TODO add here integ tests for info api: "GET /" once we have parsing code for MainResponse
+    @SuppressWarnings("unchecked")
+    public void testInfo() throws IOException {
+        MainResponse info = highLevelClient().info();
+        // compare with what the low level client outputs
+        Map<String, Object> infoAsMap = entityAsMap(adminClient().performRequest("GET", "/"));
+        assertEquals(infoAsMap.get("cluster_name"), info.getClusterName().value());
+        assertEquals(infoAsMap.get("cluster_uuid"), info.getClusterUuid());
+
+        // TODO not sure if the following always holds, low level client request we compare to might hit different node?
```

Comment:
I think these tests are run against a single node cluster, but yes it doesn't necessarily hold. just check that the node name is there? Maybe you want to use ObjectPath (although it will do the same internally)